index.js

```diff
@@ -91,16 +84,26 @@ export default function pThrottle({limit, interval, strict, onDelay}) {
 			});
 		};
 
-		throttled.abort = () => {
+		const aborted = () => {
 			for (const timeout of queue.keys()) {
 				clearTimeout(timeout);
-				queue.get(timeout)(new AbortError());
+				queue.get(timeout)(signal.reason);
 			}
 
 			queue.clear();
 			strictTicks.splice(0, strictTicks.length);
 		};
 
+		if (signal) {
+			signal.throwIfAborted();
+			signal.addEventListener('abort', aborted, {once: true});
+
+			const registry = new FinalizationRegistry(() => {
+				signal.removeEventListener('abort', aborted);
+			});
+			registry.register(signal, 'signal');
```

Comment:
This is incorrect. It's the `throttled` function that needs to be the target.

And the registry should be defined at the top level:

```js
const registry = new FinalizationRegistry(({signal, aborted}) => {
    signal.removeEventListener('abort', aborted);
});
```